test/cmd/builds.sh

```diff
@@ -12,16 +12,64 @@ os::log::install_errexit
 url=":${API_PORT:-8443}"
 project="$(oc project -q)"
 
+# expect_build_output_spec runs `oc new-build` with given arguments and expects
+# a build config with a given name to be created with a given build output spec
+function os::cmd::builds::expect_build_output_spec() {
```

Comment:
I made it break:
```
[vagrant@openshiftdev origin]$ git diff
diff --git a/test/cmd/builds.sh b/test/cmd/builds.sh
index d1a10e7..038a611 100755
--- a/test/cmd/builds.sh
+++ b/test/cmd/builds.sh
@@ -39,7 +39,7 @@ os::cmd::expect_success 'oc get bc/test'
 os::cmd::builds::expect_build_output_spec \
        "--dockerfile=\$'FROM centos:7\nRUN yum install -y httpd'" \
        'centos' \
-       'ImageStreamTag centos:latest'
+       'ImageStreamTag centos:latest_wrong'
 
 # Build from Dockerfile with output to DockerImage
 os::cmd::builds::expect_build_output_spec \
```

And ran the tests:
```
[vagrant@openshiftdev origin]$ ./hack/test-cmd.sh test/cmd/builds.sh 
<...>
++ test/cmd/builds.sh
Now using project "default" from context named "default/127-0-0-1:28443/system:admin" on server "https://127.0.0.1:28443".
Now using project "cmd-builds" on server "https://127.0.0.1:28443".
SUCCESS: oc new-build openshift/ruby-20-centos7 https://github.com/openshift/ruby-hello-world.git
SUCCESS: oc get bc/ruby-hello-world
SUCCESS: cat "${OS_ROOT}/Dockerfile" | oc new-build -D - --name=test
SUCCESS: oc get bc/test
SUCCESS: oc delete all --all
SUCCESS: oc new-build --dockerfile=$'FROM centos:7\nRUN yum install -y httpd'
FAILURE: oc get bc/centos --template '{{with .spec.output.to}}{{.kind}} {{.name}}{{end}}': the output content test failed
Standard output from the command:
ImageStreamTag centos:latestThere was no error output from the command.
!!! Error in test/cmd/../../hack/cmd_util.sh:145
        'return 1' exited with status 1
Call stack:
        1: test/cmd/../../hack/cmd_util.sh:145 os::cmd::expect_success_and_text(...)
        2: test/cmd/builds.sh:28 os::cmd::builds::expect_build_output_spec(...)
        3: test/cmd/builds.sh:40 main(...)
Exiting with status 1
!!! Error in ./hack/test-cmd.sh:284
        '${test}' exited with status 1
Call stack:
        1: ./hack/test-cmd.sh:284 main(...)
Exiting with status 1
[FAIL] !!!!! Test Failed !!!!
<...>
```


With colors and all :+1: 


To compare, I made it break at a different place:
```
[vagrant@openshiftdev origin]$ git diff
diff --git a/test/cmd/builds.sh b/test/cmd/builds.sh
index d1a10e7..37effa1 100755
--- a/test/cmd/builds.sh
+++ b/test/cmd/builds.sh
@@ -33,7 +33,7 @@ function os::cmd::builds::expect_build_output_spec() {
 os::cmd::expect_success 'oc new-build openshift/ruby-20-centos7 https://github.com/openshift/ruby-hello-world.git'
 os::cmd::expect_success 'oc get bc/ruby-hello-world'
 os::cmd::expect_success 'cat "${OS_ROOT}/Dockerfile" | oc new-build -D - --name=test'
-os::cmd::expect_success 'oc get bc/test'
+os::cmd::expect_success 'oc get bc/test2'
 
 # Build from Dockerfile with output to ImageStreamTag
 os::cmd::builds::expect_build_output_spec \
```

And the test output:
```
[vagrant@openshiftdev origin]$ ./hack/test-cmd.sh test/cmd/builds.sh 
<...>
++ test/cmd/builds.sh
Now using project "default" from context named "default/127-0-0-1:28443/system:admin" on server "https://127.0.0.1:28443".
Now using project "cmd-builds" on server "https://127.0.0.1:28443".
SUCCESS: oc new-build openshift/ruby-20-centos7 https://github.com/openshift/ruby-hello-world.git
SUCCESS: oc get bc/ruby-hello-world
SUCCESS: cat "${OS_ROOT}/Dockerfile" | oc new-build -D - --name=test
FAILURE: oc get bc/test2: the command returned the wrong error code
There was no output from the command.
Standard error from the command:
Error from server: buildconfig "test2" not found
!!! Error in test/cmd/../../hack/cmd_util.sh:145
        'return 1' exited with status 1
Call stack:
        1: test/cmd/../../hack/cmd_util.sh:145 os::cmd::expect_success(...)
        2: test/cmd/builds.sh:36 main(...)
Exiting with status 1
!!! Error in ./hack/test-cmd.sh:284
        '${test}' exited with status 1
Call stack:
        1: ./hack/test-cmd.sh:284 main(...)
Exiting with status 1
[FAIL] !!!!! Test Failed !!!!
<...>
```

In both cases we get a reasonable stack trace, a good indication of which line failed, and the exact command that failed. Agree?